packages/@okta/vuepress-theme-prose/components/Sidebar.vue

```diff
@@ -1,43 +1,54 @@
 <template>
   <aside class="landing-navigation">
     <ul class="landing">
-      <SidebarItem v-for="link in navigation" :key="link.title" :link="link" />
+      <ul class="sections">
+        <SidebarItem
+          v-for="link in navBarItems"
+          :key="link.title"
+          :link="link"
+        />
+      </ul>
     </ul>
   </aside>
 </template>
 
 <script>
 import _ from "lodash";
-import { getGuidesInfo, guideFromPath } from "../util/guides";  
+import { getGuidesInfo, guideFromPath } from "../util/guides";
 
 export default {
   name: "Sidebar",
-  inject: ['appContext'],
+  inject: ["appContext"],
   components: {
     SidebarItem: () => import("../components/SidebarItem.vue")
   },
+  computed: {
+    navigation() {
+      return (this.$page.redesign
+        ? this.getNewNavigation()
+        : this.getNavigation() || []
+      ).map(nav => {
+        this.addStatesToLink(nav);
+        return nav;
+      });
+    }
+  },
   data() {
     return {
-      usingFile: false
+      usingFile: false,
+      navBarItems: []
     };
   },
   mounted() {
-    if(!this.appContext.isInMobileViewport) {
+    if (!this.appContext.isInMobileViewport) {
       this.handleScroll();
       window.addEventListener("scroll", this.handleScroll);
     }
+    this.navBarItems = this.navigation;
```

Comment:
Is this necessary, or can we we directly access the computed `navigation`?

Reply:
if we will directly access navigation it will be triggering each time we click on navbar item

Reply:
Ah I see. Then we can directly inline the computation method in here rather than do a reassignment, e.g.
```js
this.navBarItems = (this.getNavigation() || []).map(/* the rest of the code */;
```

Reply:
Seems @ pradeepdewda-okta comment makes sense. Such changes will break some logic related to switching between vue / react / angular documentation by using [select element] (https://developer.okta.com/docs/guides/sign-into-spa/angular/add-signin-button /). So here when we switching between frameworks rout path is changing and navbar should be rerendered not lose the selected item. So probably a current solution with permanent rendering because of directly using `navigation` is not perfect but it will not break anything. And after that, we can create a task for using a more elegant solution in case of switching between different frameworks in documentation pages

Reply:
Ok, we can leave this as-is if it's needed for functionality